tools.py

```diff
@@ -79,34 +133,51 @@ def add_vertex(self, run_id, pose_id, next_run_id, next_pose_id, next_transform,
             self.matches[v.next_id].append(v)
             self.vertices[v.vertex_id] = v
         else:
-            print("Teach vertex {0} not found in graph so vertex {1} was not added.".format(v.next_id, v.vertex_id))
+            print("Warning: teach vertex {0} not found in graph so vertex {1} was not added.".format(v.next_id, v.vertex_id))
 
-    # Iterate through teach vertices to add previous indices
     def add_prev_ids(self):
+        """Iterates over all the teach vertices and adds the index of the previous vertex as an 
+           attribute to each one.
+        """
         for v_id in self.vertices:
             v = self.get_vertex(v_id)
             if v.teach and v.next_id in self.vertices:
                 self.vertices[v.next_id].prev_id = v_id
 
-    # Add new repeat run to graph
     def add_run(self, run_file):
+        """Add a new repeat run to the graph.
+
+        Args:
+            run_file (str): Path to file containing spatial transformations for the repeat run.
+        
+        """
         transforms_spatial = np.loadtxt(run_file, delimiter=",")
         for row in transforms_spatial:
             transform = Transform(np.array([row[4:7], row[8:11], row[12:15]]), np.array([row[7], row[11], row[15]]))
             self.add_vertex(row[0], row[1], row[2], row[3], transform, False)
 
-    # Add image timestamps to each vertex
-    def add_timestamps(self, time_files):
-        for run in time_files:
-            run_times = np.loadtxt(time_files[run], delimiter=",")
+    def add_timestamps(self, im_time_files):
+        """Add image timestamps to vertices.
+
+        Args:
+            im_time_files (list of str): Paths to files containing image time stamps for runs.            
+        """
+        for run in im_time_files:
+            run_times = np.loadtxt(im_time_files[run], delimiter=",")
             for row in run_times:
                 if (run, int(row[0])) in self.vertices:
                     self.vertices[(run, row[0])].timestamp = float(row[1]) * 10**-9
                 else:
                     print("Warning: attempted to add timestamp for vertex ({0}, {1}) not in graph.".format(run, int(row[0])))
 
     # Add associated GPS measurement and its timestamp to vertices where available
```

Comment:
Couple places where my brief descriptions are still in, should probably remove for consistency